src/AggregateFunctions/AggregateFunctionCramersV.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <AggregateFunctions/UniqVariadicHash.h>
+#include <Common/HashTable/HashMap.h>
+#include <DataTypes/DataTypesNumber.h>
+#include "Core/DecimalFunctions.h"
+
+
+namespace DB
+{
+    using Map1 = HashMap<UInt64, UInt64>;
+    using Map2 = HashMap<UInt64, UInt64>;
+    using Map12 = HashMap<UInt128, UInt64>;
+
+
+    struct AggregateFunctionCramersVData
+    {
+        UInt64 size_of_column = 0;
+        Map1 count_of_value1;
+        Map2 count_of_value2;
+        Map12 count_of_pairs;
+        Float64 getResult() const
+        {
+            UInt64 q = std::min(count_of_value1.size(), count_of_value2.size());
+            if (q == 1) {
+                return 1;
+            }
+            Float64 chi_squared = 0;
+            for (const auto & cell : count_of_pairs) {
+                UInt128 hash_pair = cell.getKey();
+                UInt64 hash1 = (hash_pair << 64 >> 64);
```

Comment:
Usually people just write `UInt64 hash1 = hash_pair;`, it will automatically truncate to a needed bitness